src/clj/game/core/rules.clj

```diff
@@ -460,18 +460,6 @@
          (continue-ability state side trash-effect moved-card (list cause))))
      (effect-completed state side eid))))
 
-(defn- resolve-trash
-  ([state side eid card args] (resolve-trash state side eid card eid args))
-  ([state side eid {:keys [zone] :as card} oid
-   {:keys [cause suppress-event] :as args}]
-  (if (and (not suppress-event)
-           (not= (last zone) :current)) ; Trashing a current does not trigger a trash event.
```

Comment:
You don't replicate this check for currents in your new code, is it still needed?

Reply:
I asked in the nisei rules chat, and they said that right now, no it's not, but that might change in the future. Given that all of the tests pass, I don't think anything is currently changed because of this, but we'll see once this hits live. Thankfully, it's a _very_ easy fix to re-add.

Reply:
As seen by #4529, I am proven wrong, lol. I'll have to figure out a fix without reverting this PR.